astropy/nddata/nduncertainty.py

```diff
@@ -156,26 +174,50 @@ def propagate_divide(self, other_nddata, result_data):
             The resulting uncertainty
         """
 
+        self._raise_not_implemented_arithmetic('Dividing')
+
+
+class CustomUncertainty(NDUncertainty):
+    """
+    A class for custom uncertainties
+
+    Parameters
+    ----------
+
+    array: ~np.ndarray
+
+    uncertainty_type: object
+        label to specify what kind of uncertainty you have (str is common)
```

Comment:
really? I think we only require that it `hasattr` uncertainty type. but sure we can put `~str` there.